CMakeLists.txt

```diff
@@ -577,3 +624,101 @@ add_executable (math-matrix_4x4.test test/math/matrix_4x4.cpp)
 target_compile_definitions (math-matrix_4x4.test PRIVATE "-DBOOST_TEST_MODULE=\"math\"")
 target_link_libraries (math-matrix_4x4.test Boost::unit_test_framework noggit::math)
 add_test (NAME math-matrix_4x4 COMMAND $<TARGET_FILE:math-matrix_4x4.test>)
+
+include (FetchContent)
+
+# Dependency: StormLib
+
+FetchContent_Declare (storm
+  GIT_REPOSITORY https://github.com/ladislav-zezula/StormLib.git
+)
+FetchContent_GetProperties (storm)
+if (NOT storm_POPULATED)
+  message (STATUS "Installing StormLib...")
+  FetchContent_Populate (storm)
+endif()
+add_subdirectory (${storm_SOURCE_DIR} ${storm_BINARY_DIR} EXCLUDE_FROM_ALL)
+set (storm_warning_disable_flags "")
+add_compiler_flag_if_supported (storm_warning_disable_flags -Wno-implicit-function-declaration)
+target_compile_options (storm PRIVATE ${storm_warning_disable_flags})
+
+target_link_libraries (noggit
+  storm
+)
+
+if (NOGGIT_WITH_SCRIPTING)
+  # The macros to define and use modules are missing prototypes but also don't work well with anon namespaces/static.
+  set (dascript_loader_warning_flags "")
+  add_compiler_flag_if_supported (dascript_loader_warning_flags -Wno-missing-prototypes)
+  if( NOT dascript_loader_warning_flags STREQUAL "")
+    set_source_files_properties ("src/noggit/scripting/script_loader.cpp" PROPERTIES COMPILE_FLAGS ${dascript_loader_warning_flags})
+  endif()
+
+  # Dependency: json.hpp
+  FetchContent_Declare (json
+    GIT_REPOSITORY https://github.com/ArthurSonzogni/nlohmann_json_cmake_fetchcontent
+    GIT_TAG v3.9.1
+  )
+  FetchContent_GetProperties (json)
+  if (NOT json_POPULATED)
+    message (STATUS "Installing json.hpp...")
+    FetchContent_Populate (json)
+  endif()
+  add_subdirectory (${json_SOURCE_DIR} ${json_BINARY_DIR} EXCLUDE_FROM_ALL)
+
+  # Dependency: lodepng
+  FetchContent_Declare (lodepng
+    GIT_REPOSITORY https://github.com/lvandeve/lodepng.git
+  )
+  FetchContent_GetProperties (lodepng)
+  if (NOT lodepng_POPULATED)
+    message (STATUS "Installing lodepng...")
+    FetchContent_Populate (lodepng)
+  endif()
+  add_library (lodepng "${lodepng_SOURCE_DIR}/lodepng.cpp")
+
+  # Dependency: FastNoise2
+  FetchContent_Declare (fastnoise2
+    GIT_REPOSITORY https://github.com/tswow/FastNoise2.git
+  )
+  FetchContent_GetProperties (fastnoise2)
+  if (NOT fastnoise2_POPULATED)
+    message (STATUS "Installing FastNoise2... (big repo, large download)")
+    FetchContent_Populate (fastnoise2)
+  endif()
+  add_subdirectory (${fastnoise2_SOURCE_DIR} ${fastnoise2_BINARY_DIR} EXCLUDE_FROM_ALL)
+  # hack, but something's wrong in FastNoise2s CMakeLists
+  target_include_directories (FastNoise PRIVATE ${fastnoise2_SOURCE_DIR}/include)
+  target_include_directories (NoiseTool PRIVATE ${fastnoise2_SOURCE_DIR}/include)
+
+  # Dependency: daScript
+  FetchContent_Declare (dascript
+    GIT_REPOSITORY https://github.com/ihm-ts/daScript.git
+    # don't fetch submodules, see https://gitlab.kitware.com/cmake/cmake/-/issues/20579
+    GIT_SUBMODULES "src"
```

Comment:
this generates `error: pathspec 'src' did not match any file(s) known to git` with git 2.24 when reconfiguring :(

The cmake bug was apparently fixed in 3.18, which is July 2020. That is recent, but I'd probably prefer forcing people to use that over having this pain of working around a known bug. 

Opinions?

Reply:
I agree, it's better to upgrade